packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2385,6 +2384,14 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
           }
         });
 
+    void _removeEntryIfNeeded() {
+      if (!isPersistent && !removedEntry) {
+        assert(entry != null);
+        removedEntry = true;
```

Comment:
Move this flag(`removedEntry`) forward so as not to trigger `LocalHistoryEntry.onRemove`, otherwise `_StandardBottomSheet.onClosing` will be triggered twice when using the gesture remove the sheet, which is strange behavior.